src/util/effect.ts

```diff
@@ -0,0 +1,16 @@
+/**
+ * An custom implementation of a Error Cause.
+ *
+ * See https://github.com/tc39/proposal-error-cause
+ */
+export class Effect extends Error {
```

Comment:
After reading the TC39 proposal, I think this should be `addCause(new Error('Couldn't get stuff'), innerFetchError)`. This would add the `innerFetchError` to the error as a `cause` property, and edit the stack trace to match.

That way, one could do `thow addCause(new TypeError('client sent wrong data', cleanerError)`, so it's a `TypeError` instance, which happens to have a deeper cause. The TC39 proposal was explicitly opposed to making a subclass.